### PR TITLE
Add a new gatsby-cli package

### PR DESCRIPTION
**Pull request by KyleAMathews**
Once landed, users should install this globally instead of the Gatsby package.

### CODE REVIEW

**Comment by sebastienfi** [comment]
Should this be modified too ? https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-dev-cli

**Comment by KyleAMathews** [comment]
No — this is for everyone to use. It's a replacement for installing the `gatsby` package globally e.g. running `gatsby new`. `gatsby-dev-cli` is for developing Gatsby the project.

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit d8e6b6c3dc2e6e04d46a7b7939a24acad3828948

https://app.netlify.com/sites/using-glamor/deploys/598245bd0752d007b92d1ce1

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit c7e1e63689a851170470908eed3c10d60bec0806

https://app.netlify.com/sites/using-glamor/deploys/598249c90752d053f62d1c92

File: packages/gatsby/src/bin/cli.js [hunk 1]
```diff
@@ -16,6 +17,18 @@ process.on(`unhandledRejection`, error => {
 
 const defaultHost = `localhost`
 
+let inGatsbySite = false
+let localPackageJSON
+try {
+  localPackageJSON = require(path.resolve(`./package.json`))
+  if (localPackageJSON.dependencies && localPackageJSON.dependencies.gatsby) {
```
Comment (madbence) [inline packages/gatsby/src/bin/cli.js]: @KyleAMathews is there is a reason why `devDependencies` are not checked? in our setup `gatsby` is a dev dep, and this change broke our builds :crying_cat_face: .

Comment (KyleAMathews) [inline packages/gatsby/src/bin/cli.js]: None :-) I just didn't think about that as I've never made gatsby a devDependency but can now there's plenty of use cases where that makes sense. I'll merge https://github.com/gatsbyjs/gatsby/pull/1738 and make a new release.

**Pull request merged**